### DISCUSSION TITLE
What is the resolution order for `just <recipe>` invoked from another recipe?

### DISCUSSION THREAD

**Discussion by indigoviolet** [casey/just#1541] [Q&A]
Is it guaranteed to be equivalent to `just -f justfile() <recipe>`?  

I guess not - it is just running a shell command, so if I `cd <dir> && just <recipe>`, then the resolution would be relative to `<dir>`.

Similar to `[no-cd]`, it would be useful to have a directive that fixes recipe resolution relative to the current invocation directory.

**Comment by casey** [comment]
Running `just` recursively inside of a recipe is exactly the same as running it from the shell. Recipes are run, if I recall correctly, in the order in which they are given on the command line, unless there are interdependencies which prevent that.
